docs/metrics/index.rst

```diff
@@ -0,0 +1,27 @@
+.. _metrics:
+
+==================
+Metrics Collection
+==================
+
+Timeseries metrics my be optionally collected and exported from the server.
+This feature is dissabled by default but can be enabled by passing `--enable-metrics`
+when envoking the server.
+
+The metrics offer insight into how the server is processing messages. Each subsystem publishes
+its own set of metrics. These are timeserises data so they are mostly counters of processed
+elements.
+
+We use the Prometheus metrics format and the metrics are exposed by an HTTP server which by default
+run on for 9215. The implementation is fully complient with Prometheus scraping, so you need only
+point your Prometheus server to the exposed metrics to scrape them.
+
+When using the feature, you need make a directory available which will be used to collect and store
+local metrics from the valious system processes. The server will need rights to write to this
+directory. A directory in ``/tmp`` is fine. Once created you will need to set the
+``prometheus_multiproc_dir`` environment variable which will be read by the server on startup.
+
+.. code-block:: bash
+
+    $ mkdir /tmp/prom
+    $ export prometheus_multiproc_dir=/tmp/prom
```

Comment:
As this is required I think it would be best to add a default, and create the default directory if not present. I would use something that no other program would use by default, i.e `/tmp/_napalm_logs_metrics` or something similar.

Reply:
So I have been working this out and it turns out to be a little complicated. The prometheus client library is checking for the existence of `prometheus_multiproc_dir` on import (see https://github.com/prometheus/client_python/blob/2783ff82991a34caa6f8fd855a2b3b8f218daec0/prometheus_client/core.py#L505) which means to do this correctly, I would need to import the library _after_ setting the env var. This does not seem very clean to me. Thoughts?

Reply:
It looks like you can set `path` when you initiate the class:

```
diff --git a/napalm_logs/base.py b/napalm_logs/base.py
index 5f8800e..2c1573b 100644
--- a/napalm_logs/base.py
+++ b/napalm_logs/base.py
@@ -138,7 +138,7 @@ class NapalmLogs:
                     if f.endswith(".db"):
                         os.remove(os.path.join(path, f))
             registry = CollectorRegistry()
-            multiprocess.MultiProcessCollector(registry)
+            multiprocess.MultiProcessCollector(registry, path='/tmp/_napalm_logs_prom/')
             start_http_server(
                 port=self.metrics_port,
                 addr=self.metrics_address,
```

You will need to check / make the dir first.

Reply:
Oh, I can see the issue here, let me have a think.

Reply:
What if we read from default or the config file, then set the env variable from [cli.py](https://github.com/napalm-automation/napalm-logs/blob/develop/napalm_logs/scripts/cli.py)

This way the env variable will be set before we import.

Reply:
I think we will still run into a recursive import based on `napalm_logs.base` but I will play with it.

Reply:
What if we move the default for this into either `napalm_logs/scripts/__init__.py` or `napalm_logs/__init__.py`? With `napalm_logs/__init__.py` it would have to appear before `from napalm_logs.base import NapalmLogs`. We could se the env var to the default and then later ovewrite it if passed via cli or config. This get's us past the import. Theprometheus library is only checking that the var exsits. It does not care what the value is until it is actually needed.

Reply:
Well, I guess the default doesn't _have_ to be set there. We only need to ensure the env var it set before the import. After the import, we can change it. We could have it so we always overwrite the value to either the proper default of the user-defined value.

Reply:
Actually it will have to be `napalm_logs/__init__.py` because of `from napalm_logs.base import NapalmLogs` and it will of couse have to come before that. 
```python
# -*- coding: utf-8 -*-
'''
napalm-logs package init
'''
from __future__ import absolute_import
from __future__ import unicode_literals

import pkg_resources
import os

# For metrics collection, the prometheus_client library
# requires that the env var `prometheus_multiproc_dir`
# be set before start up. At startup, it doesn't matter
# what the value is, and we always overwrite it to either
# the poperly defined default or the user propvided value.
os.environ['prometheus_multiproc_dir'] = "/tmp/napalm_logs_metrics"

# Import napalm-logs pkgs
from napalm_logs.base import NapalmLogs

try:
    __version__ = pkg_resources.get_distribution('napalm-logs').version
except pkg_resources.DistributionNotFound:
    __version__ = "Not installed"


__all__ = ('NapalmLogs', '__version__')
```
This seems to work fine.